### PR TITLE
Add unfollow modal (optional)

### PR DESCRIPTION
**Pull request by ykzts**
I don't want to unfollow friends by mistake.

### CODE REVIEW

File: config/locales/simple_form.en.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ en:
         setting_default_sensitive: Always mark media as sensitive
         setting_delete_modal: Show confirmation dialog before deleting a toot
         setting_system_font_ui: Use system's default font
+        setting_unfollow_modal: Show confirmation dialog before unfollow an account
```
Comment (Gargron) [inline config/locales/simple_form.en.yml]: "Before unfollowing someone"

File: app/javascript/mastodon/components/account.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ export default class Account extends ImmutablePureComponent {
   static propTypes = {
     account: ImmutablePropTypes.map.isRequired,
     me: PropTypes.number.isRequired,
+    unfollowModal: PropTypes.bool,
```
Comment (Gargron) [inline app/javascript/mastodon/components/account.js]: I think this component doesn't need to be aware of this.

File: app/javascript/mastodon/features/account_timeline/components/header.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export default class Header extends ImmutablePureComponent {
   static propTypes = {
     account: ImmutablePropTypes.map,
     me: PropTypes.number.isRequired,
+    unfollowModal: PropTypes.bool,
```
Comment (Gargron) [inline app/javascript/mastodon/features/account_timeline/components/header.js]: I think this component doesn't need to be aware of this.

**Review by Gargron — CHANGES_REQUESTED** [review]
👍

File: config/locales/simple_form.en.yml (hunk 1 above)
Comment (ykzts) [inline config/locales/simple_form.en.yml]: thx! 9687d82

File: app/javascript/mastodon/features/account_timeline/components/header.js (hunk 3 above)
Comment (ykzts) [inline app/javascript/mastodon/features/account_timeline/components/header.js]: thx! 8724533

**Review by Gargron — APPROVED** [review]

**Pull request merged**